src/main/msc/emfat.c

```diff
@@ -527,7 +527,7 @@ void fill_entry(dir_entry *entry, const char *name, uint8_t attr, uint32_t clust
         l2 = l2 > FILE_NAME_EXTN_LEN ? FILE_NAME_EXTN_LEN : l2;
     }
 
-    memset(entry->name, ' ', FILE_NAME_SHRT_LEN + FILE_NAME_EXTN_LEN);
+    memset(entry->name, ' ', FILE_NAME_SHRT_LEN);
```

Comment:
Not sure this is correct - this `memset()` sets `name` and `extn` to ` `, as it should, as the subsequent `memset()` calls can write less than the full length of the field.
This is done in a clumsy way, and should be fixed, but not blanking `extn` before writing does not seem to be the correct approach.

Reply:
I think it does not segfault currently, because the page fault is triggered at 17 bytes,
so `memset(entry->name, ' ', 16 + 1) `does fault and `memset(entry->name, ' ', 16)` or < 16 does not. However, since `entry->name` has size `FILE_NAME_SHRT_LEN `the call
`memset(entry->name, ' ', FILE_NAME_SHRT_LEN + FILE_NAME_EXTN_LEN)`
results in an access out of bound.

Also my emacs flycheck  is complaining:

Buffer is accessed out of bounds: [bufferAccessOutOfBounds]
'memset' will always overflow; destination buffer has size 8, but size argument is 11

Reply:
This is still missing the setting of `entry->extn` to `   `.

Reply:
Done